src/back/glsl/mod.rs

```diff
@@ -1127,6 +1134,14 @@ impl<'a, W: Write> Writer<'a, W> {
                         " {}",
                         &self.names[&NameKey::StructMember(handle, idx as u32)]
                     )?;
+                    // Do not write size for types with builtin bindings
+                    // implicitly sized arrays cannot be assigned
+                    if let Some(Binding::BuiltIn(_)) = member.binding {
```

Comment:
@kvark I missed one thing. Real problem fixed in this part of PR.

```glsl
struct type20 {
    vec2 member;
    vec4 member1;
    vec4 gen_gl_Position1;
    float gen_gl_PointSize1;
    float gen_gl_ClipDistance1[1];
};

gl_ClipDistance = type20(f_tex_pos, f_color, perVertexStruct.gen_gl_Position, perVertexStruct.gen_gl_PointSize, perVertexStruct.gen_gl_ClipDistance).gen_gl_ClipDistance1;
```

```bash
error: implicitly sized arrays cannot be assigned
 shader compilation failed: Vertex
```

Other problems in the log are warnings. I think they are produced [here](https://github.com/mesa3d/mesa/blob/43dd023bd1eb23a5cdb1470c6a30595c3fbf319a/src/compiler/glsl/ast_function.cpp#L318) in mesa.

Reply:
We had a similar problem in MSL backend. There is code now that just sub-initializes the arrays on return. You can see it in "quad-vert.msl":
```
const auto _tmp = type10 {v_uv, perVertexStruct.gl_Position, perVertexStruct.gl_PointSize, perVertexStruct.gl_ClipDistance};
    return main2Output { _tmp.member, _tmp.gl_Position1, _tmp.gl_PointSize1, {_tmp.gl_ClipDistance1.inner[0]} };
```

Reply:
I decided to skip gl_CullDistance, gl_ClipDistance in a generated shader for ES target. We should find a better way how we can handle them.

For example:
```glsl
#version 450

void main() {
    gl_Position = vec4(1.0, 1.0, 1.0, 1.0);
    gl_ClipDistance[0] = 0.0f;
}
```

Generates to:
```glsl
#version 320 es

precision highp float;

struct type8 {
    vec4 gen_gl_Position1;
    float gen_gl_PointSize1;
    float gen_gl_ClipDistance1[1];
    float gen_gl_CullDistance1[1];
};

struct gen_gl_PerVertex_block_0 {
    vec4 gen_gl_Position;
    float gen_gl_PointSize;
    float gen_gl_ClipDistance[1];
    float gen_gl_CullDistance[1];
} perVertexStruct;

void main1() {
    perVertexStruct.gen_gl_Position = vec4(1.0, 1.0, 1.0, 1.0);
    perVertexStruct.gen_gl_ClipDistance[0] = 0.0;
    return;
}

void main() {
    main1();
    type8 _tmp_return = type8(perVertexStruct.gen_gl_Position, perVertexStruct.gen_gl_PointSize, perVertexStruct.gen_gl_ClipDistance, perVertexStruct.gen_gl_CullDistance);
    gl_Position = _tmp_return.gen_gl_Position1;
    gl_PointSize = _tmp_return.gen_gl_PointSize1;
    gl_ClipDistance = _tmp_return.gen_gl_ClipDistance1;
    gl_CullDistance = _tmp_return.gen_gl_CullDistance1;
    return;
}
```